### PR TITLE
#79 i18n

### PR DESCRIPTION
**Pull request by Tibesnoff**
Added i18n to app

### CODE REVIEW

File: app/src/Components/Screens/Home/Home.tsx [hunk 1]
```diff
@@ -32,23 +34,28 @@ const HomeScreen: React.FC<HomeScreenProps> = () => {
...
           <PrimaryText className="text-left">
-            Specific Places: {item.category?.specific_places.join(', ')}
+            {`${i18n.t(
+              'Wallet.Specific',
+            )}: ${item.category?.specific_places.join(', ')}`}
           </PrimaryText>
         )}
         <PrimaryText className="text-left">
-          Initial Percentage: {item.initial_percentage}
+          {`${i18n.t('Wallet.Initial')} ${i18n.t('Wallet.Percentage')}:
```
Comment (alexanderjhughes) [inline app/src/Components/Screens/Home/Home.tsx:47]: Hmmm why do we have it like this? Why not just initial_percentage, english is very different then other languages and in some languages percentage might be before initial. It should just be a initial_percnetage option in the translation.

File: app/src/Components/Screens/Home/Home.tsx [hunk 2]
```diff
@@ -32,23 +34,28 @@ const HomeScreen: React.FC<HomeScreenProps> = () => {
...
         </PrimaryText>
-        <PrimaryText>Initial Limit: {item.initial_limit}</PrimaryText>
+        <PrimaryText>{`${i18n.t('Wallet.Initial')} ${i18n.t('Wallet.Limit')}: 
+          ${item.initial_limit}`}</PrimaryText>
         <PrimaryText>
-          Term Length (time until limit resets): {item.term_length_months} Month
-          {item.term_length_months > 1 ? 's' : ''}
+          {`${i18n.t('Wallet.Term')}: ${item.term_length_months} ${
+            item.term_length_months > 0 ? i18n.t('Wallet.Month') : ''
+          }${item.term_length_months > 1 ? i18n.t('Wallet.S') : ''}`}
```
Comment (alexanderjhughes) [inline app/src/Components/Screens/Home/Home.tsx:55]: For things like this, see [i18n Interpolation](https://www.i18next.com/translation-function/interpolation).
You can basically pass things into the string so you can do
```
Wallet.TermSingular = "Term Length: %months% Month"
Wallet.TermPlural = "Term Length: %months% Months"
```
much easier then to translate into different languages. Translating just a single 'S' does nothing when that is not how other languages speak.

File: app/src/Components/Screens/Home/Home.tsx [hunk 3]
```diff
@@ -70,19 +78,24 @@ const HomeScreen: React.FC<HomeScreenProps> = () => {
...
-        Hello {userProfile.username}
+        {i18n.t('Common.Hello', { username: userProfile.username })}
       </TitleText>
       {topCard ? (
         <>
           <View className="w-full justify-center items-center h-1/3">
-            <PrimaryText className="text-center text-xl mt-10">{`You're at ${address?.displayName.text}\n We suggest you use the following card`}</PrimaryText>
+            <PrimaryText className="text-center text-xl mt-10">{`${i18n.t(
+              'Home.At',
+              { address: address?.displayName.text },
```
Comment (alexanderjhughes) [inline app/src/Components/Screens/Home/Home.tsx:88]: Use this but for months

File: app/src/Components/Screens/Profile/Profile.tsx [hunk 4]
```diff
@@ -56,17 +49,24 @@ const ProfileScreen: React.FC<ProfileScreenProps> = () => {
...
         <PrimaryText className="text-2xl font-bold mt-20">
-          You have {cardCount} card
-          {cardCount > 1 ? 's' : ''}
+          {`${i18n.t('Profile.Have')} ${cardCount} ${
+            cardCount > 1 ? i18n.t('Profile.Cards') : i18n.t('Profile.Card')
+          }`}
         </PrimaryText>
         <PrimaryText className="text-2xl font-bold">
-          {`You have ${rewardCount} reward${rewardCount !== 1 ? 's' : ''}`}
+          {`${i18n.t('Profile.Have')} ${rewardCount} ${
```
Comment (alexanderjhughes) [inline app/src/Components/Screens/Profile/Profile.tsx:65]: Use interpolation here so you can do:
En: "You have %cards_length" cards"
Es: "Tienes %cards_length" cartas"

File: app/src/Components/Screens/Wallet/AddCardFullForm.tsx [hunk 5]
```diff
@@ -205,20 +202,20 @@ const AddCardFullForm: React.FC = () => {
           (card.card_type !== 'Credit' && card.card_type !== 'Debit')
         ) {
           console.log('invalid card type');
-          console.log(Consts.authErrorMessages.invalidCardType);
-          addAuthError(Consts.authErrorMessages.invalidCardType);
+          console.log(Consts.authErrorMessages().invalidCardType);
+          addAuthError(Consts.authErrorMessages().invalidCardType);
```
Comment (alexanderjhughes) [inline app/src/Components/Screens/Wallet/AddCardFullForm.tsx:206]: The error messages need to be i18n as well

File: app/src/Components/Screens/Wallet/EnterBinForm.tsx
Comment (alexanderjhughes) [inline app/src/Components/Screens/Wallet/EnterBinForm.tsx:1]: This file isn't used I thought I deleted it but I guess not...

File: app/src/Components/Screens/Wallet/Wallet.tsx [hunk 6]
```diff
@@ -95,23 +97,32 @@ const WalletScreen: React.FC<WalletScreenProps> = () => {
...
-            Specific Places: {item.category?.specific_places.join(', ')}
+            {`${i18n.t(
+              'Wallet',
+              'Specific Places',
+            )}: ${item.category?.specific_places.join(', ')}`}
           </PrimaryText>
         )}
         <PrimaryText className="text-left">
-          Initial Percentage: {item.initial_percentage}
+          {`${i18n.t('Wallet.Initial')} ${i18n.t('Wallet.Percentage')}: ${
```
Comment (alexanderjhughes) [inline app/src/Components/Screens/Wallet/Wallet.tsx:111]: Use interpolation

File: app/src/Context/AuthState.tsx [hunk 7]
```diff
@@ -27,7 +26,7 @@ const AuthState: React.FC<PropsWithChildren> = ({ children }) => {
   const auth0URL = Config.REACT_APP_AUTH0_DOMAIN;
   const auth0ClientId = Config.REACT_APP_AUTH0_CLIENT_ID;
   const auth0Audience = Config.REACT_APP_AUTH0_AUDIENCE;
-  const ErrorMessages = Consts.authErrorMessages;
+  const ErrorMessages = Consts.authErrorMessages();
```
Comment (alexanderjhughes) [inline app/src/Context/AuthState.tsx:29]: Really you should be returning the error messages provided back from the backend. That was the whole point of setting up i18n on the backend.

File: app/src/Helpers/Consts.tsx
Comment (alexanderjhughes) [inline app/src/Helpers/Consts.tsx:1]: Really should replace a lot of these with error messages we get back from backend. Also, using i18n.t in this won't always be a good thing as it is not a reactive component. Just a heads up for the future.

File: app/src/Localization/README.md
Comment (alexanderjhughes) [inline app/src/Localization/README.md:1]: Good

File: app/src/Localization/en/common.json [hunk 8]
```diff
@@ -0,0 +1,141 @@
+{
+  "Common": {
+    "Back": "Back",
+    "S": "s",
```
Comment (alexanderjhughes) [inline app/src/Localization/en/common.json:4]: ew

**Review by alexanderjhughes — CHANGES_REQUESTED** [review]

File: app/src/Components/Screens/Home/Home.tsx (hunk 1 above)
Comment (Tibesnoff) [reply app/src/Components/Screens/Home/Home.tsx:47]: Ya see this is what I was talking about with Lucas. It isn't correct. I just don't know if we have the time to fix it

File: app/src/Components/Screens/Home/Home.tsx (hunk 2 above)
Comment (Tibesnoff) [reply app/src/Components/Screens/Home/Home.tsx:55]: Ya no same as above. I noticed this after I did all the work. I don't think I have time to go back and fix it.

File: app/src/Components/Screens/Wallet/AddCardFullForm.tsx (hunk 5 above)
Comment (Tibesnoff) [reply app/src/Components/Screens/Wallet/AddCardFullForm.tsx:206]: They should be

Comment (Tibesnoff) [reply app/src/Components/Screens/Wallet/AddCardFullForm.tsx:206]: I have them translate on set pretty much

File: app/src/Context/AuthState.tsx (hunk 7 above)
Comment (Tibesnoff) [reply app/src/Context/AuthState.tsx:29]: Agree no time tho for later. I am going to add a bunc h of backlog items to fix this stuff

File: app/src/Localization/en/common.json (hunk 8 above)
Comment (Tibesnoff) [reply app/src/Localization/en/common.json:4]: Yup

**Pull request merged**